src/niscope/metadata/functions_addon.py

```diff
@@ -4,4 +4,317 @@
+                'name': 'month',
+                'type': 'ViInt32'
+            }
+        ],
+        'python_name': 'get_ext_cal_last_date_and_time',
+        'returns': 'ViStatus'
+    }
+}
+
+functions_additional_get_self_cal_last_temp = {
+    'GetLastSelfCalTemp': {
+        'codegen_method': 'python-only',
+        'python_name': 'get_self_cal_last_temp',
+        'documentation': {
+            'description': 'Returns the **Temperature** during the last self calibration procedure.',
+            'note': 'The NI 4050 and NI 4060 are not supported.'
+        },
+        'method_templates': [
+            {
+                'documentation_filename': 'default_method',
+                'method_python_name_suffix': '',
+                'session_filename': 'self_cal_last_temp'
+            }
+        ],
+        'parameters': [
+            {
+                'direction': 'in',
+                'documentation': {
+                    'description': 'Identifies a particular instrument session. You obtain the **vi** parameter from niDMM_init or niDMM_InitWithOptions. The default is None.'
+                },
+                'name': 'vi',
+                'type': 'ViSession'
+            },
+            {
+                'direction': 'out',
+                'documentation': {
+                    'description': 'Returns the **temperature** during the last calibration.'
+                },
+                'name': 'temperature',
+                'type': 'ViReal64'
+            }
+        ],
+        'returns': 'ViStatus'
+    }
+}
+
+functions_additional_get_ext_cal_last_temp = {
+    'GetLastExtCalTemp': {
+        'codegen_method': 'python-only',
+        'python_name': 'get_ext_cal_last_temp',
+        'documentation': {
+            'description': 'Returns the **Temperature** during the last external calibration procedure.',
+            'note': 'The NI 4050 and NI 4060 are not supported.'
+        },
+        'method_templates': [
+            {
+                'documentation_filename': 'default_method',
+                'method_python_name_suffix': '',
+                'session_filename': 'ext_cal_last_temp'
+            }
+        ],
+        'parameters': [
+            {
+                'direction': 'in',
+                'documentation': {
+                    'description': 'Identifies a particular instrument session. You obtain the **vi** parameter from niDMM_init or niDMM_InitWithOptions. The default is None.'
+                },
+                'name': 'vi',
+                'type': 'ViSession'
+            },
+            {
+                'direction': 'out',
+                'documentation': {
+                    'description': 'Returns the **temperature** during the last calibration.'
+                },
+                'name': 'temperature',
+                'type': 'ViReal64'
+            }
+        ],
+        'returns': 'ViStatus'
+    }
+}
+
+functions_additional_get_cal_user_info = {
+    'CalFetchMiscInfo': {
```

Comment:
A lot of these are public functions that are present in the internal metadata. You should update that, then publish the new version of functions.py here.

Reply:
Internal metadata has been used to generate these functions in functions.py.